src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -838,6 +838,14 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
         debug!("looking for the `Self` type");
         let self_id = if item.is_fake() {
             None
+        // Checking if the item is a field in a variant in an enum
```

Comment:
Variants can only appear in enums.
```suggestion
        // Checking if the item is a field in an enum variant
```